packages/doenetml-worker-rust/lib-doenetml-core/src/core/state/prop/updater.rs

```diff
@@ -91,7 +93,7 @@ pub trait PropUpdaterWithCache<T: Default + Clone>: std::fmt::Debug {
 
     /// Calculate the value of the prop from the currently cached query results.
     /// Results of this function will be cached, so local caching is not needed.
-    fn calculate(&self) -> PropCalcResult<T>;
+    fn calculate(&mut self) -> PropCalcResult<T>;
```

Comment:
This `mut` does not make me happy...can we do this with interior mutability instead? (I could be talked out of this opinion, though...)

Reply:
I wasn't able to get rid of this `mut`, but I did get rid of others. We can revisit mutability patterns after we finish retooling this part of core.